### PR TITLE
SALTO-2548 fix element source filter

### PR DESCRIPTION
**Pull request by yanaise**
Fixes adapter config to correctly handle multi app scenario

---

---
_Release Notes_: 
Netsuite Adapter:
- Fixed issue where saved searches would change on fetch when in a multiple-account workspace 

---
_User Notifications_: 
_None_

### CODE REVIEW

File: packages/core/test/core/adapters/adapters.test.ts [hunk 1]
```diff
@@ -184,24 +184,39 @@ describe('adapters.ts', () => {
...
       expect(await elementsSource.has(objectType.elemID)).toBeTruthy()
+      expect(await elementsSource.has(new ElemID('d1', 'type2'))).toBeFalsy()
       expect(await elementsSource.has(new ElemID('dummy', 'type2'))).toBeFalsy()
 
+
       expect(await elementsSource.get(objectType.elemID)).toBeDefined()
+      expect(await elementsSource.get(new ElemID('d1', 'type2'))).toBeUndefined()
       expect(await elementsSource.get(new ElemID('dummy', 'type2'))).toBeUndefined()
 
+      const d1ElementsSource = result.d1?.elementsSource
```
Comment (ori-moisis) [inline packages/core/test/core/adapters/adapters.test.ts:209]: this is testing something completely different from what this test originally did and from what the test claims to do.
the test says it is checking the element source is filtered properly, but this new part doesn't check filtering at all

Comment (yanaise) [reply packages/core/test/core/adapters/adapters.test.ts:209]: Mm, do you mean I should slightly rephrase the description? I think the principle is still right.
Lines 200-207 test that no matter what elemID you give it, the element source meant for adapter with account name "dummy" will not return a type that only exists with account name "d1".
The lines 212-214 test that the element source for account name "d1" works when you query it with "dummy" (which is what is needed by the adapter), and the lines 216-218 test that the element source is still properly readOnly.
Basically this is all still a test that verifies that the elements source we give to the adapter works properly and doesn't return things it shouldn't return.

Comment (ori-moisis) [reply packages/core/test/core/adapters/adapters.test.ts:209]: you just described 3 tests in your comment
1. `it('should only return elements that belong to the relevant account', ...` (this is the original test)
2. `it('should return renamed elements when account name is different from adapter name', ...`
3. `it('should not modify elements in the origin elements source', ...`

technically it is true that we can put all unit tests in one `it('should work properly', ...` but that isn't really great for describing what behaviors and features we expect from the code under test

File: packages/core/src/core/adapters/adapters.ts [hunk 2]
```diff
@@ -123,14 +123,20 @@ export const createElemIDReplacedElementsSource = (
...
+        awu(await elementsSource.getAll()).map(async element => {
+          const ret = element.clone()
+          await updateElementsWithAlternativeAccount(
+            [ret], adapter, account, elementsSource
+          )
+          return ret
+        }),
       get: async id => {
-        const element = await elementsSource.get(createAdapterReplacedID(id, account))
+        const element = (await elementsSource.get(createAdapterReplacedID(id, account))).clone()
```
Comment (ori-moisis) [inline packages/core/src/core/adapters/adapters.ts:135]: looks like `element` might be `undefined` according to the next line here
won't this crash on trying to call `.clone()` on `undefined` in that case?

File: packages/core/test/core/adapters/adapters.test.ts [hunk 3]
```diff
@@ -182,24 +184,30 @@ describe('adapters.ts', () => {
       expect(Object.keys(result)).toEqual([serviceName])
     })
 
-    it('should return an ReadOnlyElementsSource with only the adapter elements', async () => {
-      const objectType = new ObjectType({ elemID: new ElemID(serviceName, 'type1') })
-      const result = await getAdaptersCreatorConfigs(
-        [serviceName],
+    const setupMultiAppAdapterConfig = async (): Promise<Record<string,
```
Comment (ori-moisis) [inline packages/core/test/core/adapters/adapters.test.ts:187]: nit: `jest` has a built-in mechanism for running common test setup with `beforeEach`, when a group of tests all need the same setup, using that mechanism is the more standard way to do that

**Review by ori-moisis — APPROVED** [review]
